sequencer/src/message_compat_tests.rs

```diff
@@ -0,0 +1,243 @@
+            view_number: ViewNumber::genesis(),
+            signatures: Default::default(),
+            _pd: Default::default(),
+        }),
+        GeneralConsensusMessage::TimeoutVote(TimeoutVote {
+            signature: (sender, signature.clone()),
+            data: timeout_data,
+            view_number: ViewNumber::genesis(),
+        }),
+        GeneralConsensusMessage::UpgradeProposal(Proposal {
+            data: UpgradeProposal {
+                upgrade_proposal: upgrade_data.clone(),
+                view_number: ViewNumber::genesis(),
+            },
+            signature: signature.clone(),
+            _pd: Default::default(),
+        }),
+        GeneralConsensusMessage::UpgradeVote(UpgradeVote {
+            signature: (sender, signature.clone()),
+            data: upgrade_data,
+            view_number: ViewNumber::genesis(),
+        }),
+    ];
+    let da_messages = [
+        DaConsensusMessage::DaProposal(Proposal {
+            data: DaProposal {
+                encoded_transactions: payload.encode(),
+                metadata,
+                view_number: ViewNumber::genesis(),
+            },
+            signature: signature.clone(),
+            _pd: Default::default(),
+        }),
+        DaConsensusMessage::DaVote(DaVote {
+            signature: (sender, signature.clone()),
+            data: da_data.clone(),
+            view_number: ViewNumber::genesis(),
+        }),
+        DaConsensusMessage::DaCertificate(DaCertificate {
+            data: da_data.clone(),
+            vote_commitment: da_data.commit(),
+            view_number: ViewNumber::genesis(),
+            signatures: Default::default(),
+            _pd: Default::default(),
+        }),
+        DaConsensusMessage::VidDisperseMsg(Proposal {
+            data: VidDisperseShare::from_vid_disperse(VidDisperse::from_membership(
+                ViewNumber::genesis(),
+                vid_scheme(1).disperse(payload.encode()).unwrap(),
+                &membership,
+            ))
+            .remove(0),
+            signature: signature.clone(),
+            _pd: Default::default(),
+        }),
+    ];
+    let data_messages = [DataMessage::SubmitTransaction(
+        transaction,
+        ViewNumber::genesis(),
+    )];
+
+    let seq_messages = consensus_messages
+        .into_iter()
+        .map(SequencingMessage::General)
+        .chain(da_messages.into_iter().map(SequencingMessage::Da));
+    let messages = seq_messages
+        .map(MessageKind::Consensus)
+        .chain(data_messages.into_iter().map(MessageKind::Data))
+        .map(|kind| Message { kind, sender })
+        .collect::<Vec<Message<SeqTypes>>>();
+
+    // Ensure the current serialization implementation generates the same JSON as the committed
+    // reference.
+    let actual = serde_json::to_value(&messages).unwrap();
+    let expected: Value = serde_json::from_str(include_str!("../../data/messages.json")).unwrap();
+    assert_eq!(
```

Comment:
I think we can make the this a bit more convenient for our futures selves. For example we could use https://github.com/rust-pretty-assertions/rust-pretty-assertions, with that if we compare the json strings instead of the `Value`s the output looks pretty nice and indicates where the difference is.

However the output is still pretty big, so we could also write the new JSON out to a file next to `messages.json` which makes it easier to manually diff it and update the file in the repo if desired.

Reply:
Love it

Reply:
[111fe29](https://github.com/EspressoSystems/espresso-sequencer/pull/1486/commits/111fe2947c02b705bd7675713c3997bfe1499e09)